### PR TITLE
DBZ-644 Optionally adding source column type info to field schema

### PR DESCRIPTION
**Pull request by gunnarmorling**
https://issues.jboss.org/browse/DBZ-644

Supersedes #508.

### CODE REVIEW

**Comment by gunnarmorling** [comment]
Thanks a lot for reviewing, @orrganani! This will be part of next week's release of Debezium 0.8.0.CR1.

File: debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/MySqlSourceTypeInSchemaIT.java [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+        assertThat(c2SchemaParameters).includes(entry(TYPE_NAME_PARAMETER_KEY, "MEDIUMINT"));
+
+        // variable width, name and length info
+        Map<String, String> c3aSchemaParameters = before
+                .schema()
+                .field("c3a")
+                .schema()
+                .parameters();
+
+        assertThat(c3aSchemaParameters).includes(
```
Comment (jpechane) [inline debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/MySqlSourceTypeInSchemaIT.java]: Test for precision is missing

File: debezium-core/src/main/java/io/debezium/relational/RelationalDatabaseSchema.java [hunk 2]
```diff
@@ -33,15 +34,15 @@
     private final SchemasByTableId schemasByTableId;
     private final Tables tables;
 
-    protected RelationalDatabaseSchema(String serverName, TopicSelector<TableId> topicSelector,
-            Predicate<TableId> tableFilter, Predicate<ColumnId> columnFilter, ColumnMappers columnMappers,
-            TableSchemaBuilder schemaBuilder, boolean tableIdCaseInsensitive) {
+    protected RelationalDatabaseSchema(Configuration config, String serverName, TopicSelector<TableId> topicSelector,
```
Comment (jpechane) [inline debezium-core/src/main/java/io/debezium/relational/RelationalDatabaseSchema.java]: `serverName` is a part of `Configuration` so it should not be passed separately

**Comment by jpechane** [comment]
@gunnarmorling Are the metadata kept with event flattening SMT?

File: debezium-core/src/main/java/io/debezium/relational/RelationalDatabaseSchema.java (hunk 2 above)
Comment (gunnarmorling) [reply debezium-core/src/main/java/io/debezium/relational/RelationalDatabaseSchema.java]: Ok, changed that to pass `CommonConnectorConfig` (which exposes `getLogicalName()`) instead.

**Comment by gunnarmorling** [comment]
@jpechane Ok, so the schema parameters are also kept when applying the unwrapping SMT. I've pushed one more commit related to your other suggestion on how the logical name is passed.

**Comment by jpechane** [comment]
@gunnarmorling One more question regarding the logical name? Would it be better to keep the `logicalNameField` constructor parameter and instead of it modify the field definition for Postgres to use `withDefault` with lazy constrution (`Supplier`) variant?

**Comment by gunnarmorling** [comment]
Hum, dunno. Looks overly complex to me? What would we gain by that?

**Comment by gunnarmorling** [comment]
@jpechane All looks good to me with this commit locally. Should we wait for Travis, too, I suppose?

**Comment by jpechane** [comment]
@gunnarmorling Applied, thanks! I've disabled the test - it is passing localy and on Jenkins but there are still (smaller) issues on Travis. I am pretty sure this is test environment porbalem so I switched it off to unblock path forward.

**Comment by gunnarmorling** [comment]
@jpechane Excellent, thanks! I've also resolved the JIRA issue, let's not forget about it when merging PRs. I'm going to create a follow-up JIRA issue for re-enabling that test.

**Pull request merged**